src/test/test-quota-util.c

```diff
@@ -0,0 +1,71 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <sys/mman.h>
+#include <sys/quota.h>
+
+#include "blockdev-util.h"
+#include "device-util.h"
+#include "quota-util.h"
+#include "tests.h"
+
+#define NUMINTS  (1000)
+#define FILESIZE (NUMINTS * sizeof(int))
+
+int *file_nmapped_addr(const char *path);
+
+TEST(quotactl_devnum) {
+        _cleanup_free_ char *devnode = NULL;
+        int r;
+        int fd;
+        dev_t devnum;
+        int *m_addr;
+        const char *path = "/tmp/mmapped.bin";
+
+        int id = getuid();
+        ASSERT_OK(id);
```

Comment:
I guess you want to make the test skipped when non-root user invoke this.
Then, please introduce `intro()` and use `DEFINE_TEST_MAIN_WITH_INTRO()`. See other test codes that uses the macro.